utils/src/main/java/software/amazon/awssdk/utils/async/LimitingSubscriber.java

```diff
@@ -35,19 +35,25 @@ public LimitingSubscriber(Subscriber<? super T> subscriber, int limit) {
 
     @Override
     public void onSubscribe(Subscription subscription) {
-        super.onSubscribe(subscription);
         this.subscription = subscription;
+        if (limit == 0) {
+            subscription.cancel();
+            EmptySubscriptionEnum.complete(this.subscriber);
```

Comment:
We already have [EmptySubscription](https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/pagination/async/EmptySubscription.java)  :)

Reply:
The existing EmptySubscription is in sdk-core module, and util is already a dependency of sdk-core module, so instead of using this class create a new EmptySubscription exactly same as the one in sdk-core.